// Copyright (C) 2016-2018,2021 Rodrigo Jose Hernandez Cordoba
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
package AeonGames;

import "reference.proto";

message AssemblyMsg
{
    ReferenceMsg        mesh        = 1;
	ReferenceMsg        pipeline    = 2;
	ReferenceMsg        material    = 3;
}

message ModelMsg {
	ReferenceMsg             default_pipeline         = 1;
	ReferenceMsg             default_material         = 2;
    ReferenceMsg             skeleton                 = 3;
    repeated AssemblyMsg     assembly                 = 4;
    repeated ReferenceMsg    animation                = 5;
}
